src/common/registry/storage/key.storage.ts

```diff
@@ -16,6 +16,27 @@ export class RegistryKeyStorageService {
     return await this.repository.find(where, options);
   }
 
+  async *fetchKeysByChunks(where: FilterQuery<RegistryKey>, options: any): AsyncGenerator<any, void, unknown> {
+    const batchSize = 10000;
+    let offset = 0;
+    // TODO: transaction
+    while (true) {
+      const query = this.repository.createQueryBuilder().select('*').where(where).limit(batchSize).offset(offset);
+
+      const chunk = await query.execute();
+
+      if (chunk.length === 0) {
+        break;
+      }
+
+      offset += batchSize;
```

Comment:
For better performance in db. Better and easy to use cursors instead limit&offset approach

Example:
https://www.geeksforgeeks.org/postgresql-cursor/

Reply:
MikroORM doesn't support it, I wrote it in the RFC